### PR TITLE
Tickets/dm 33632

### PR DESCRIPTION
**Pull request by teweitsai**
Initial upload.

### CODE REVIEW

**Review by pkubanek — APPROVED** [review]
The starting point of the M2 GUI. I marked some improvements, which are revolving around those topics:
- better code structure
- use of Qt provide facilities in favor of Python facilities
- questions about need for a GUI application to log anything in console (except for Qt provided messages)
- better naming

File: python/lsst/ts/m2gui/layout/layout_control.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+__all__ = ["LayoutControl"]
+
+from PySide2 import QtCore
+from PySide2.QtWidgets import QVBoxLayout
+
+from ..utils import set_button
+
+
+class LayoutControl(object):
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply python/lsst/ts/m2gui/layout/layout_control.py:30]: I think it only says: "Note: Signals should be defined only within classes inheriting from QObject. This way the signal information is added to the class QMetaObject structure.". In my test, the `LayoutContol` does not need to inherit from this class, but the `SignalControl` does. Thanks!

File: python/lsst/ts/m2gui/signal/signal_control.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+__all__ = ["SignalControl"]
+
+from PySide2 import QtCore
+
+
+class SignalControl(QtCore.QObject):
+    """Control signal to send the information that the commandable SAL
+    component (CSC) is the commander or not."""
+
+    is_csc_commander = QtCore.Signal(bool)
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply python/lsst/ts/m2gui/signal/signal_control.py:31]: I prefer to put all the signal classes into a submodule under the `signal/` directory for the future's documentation and maintenance. It is easier to search which class is using this signal by *Sublime Text* editor. Thanks!

File: bin/run_m2gui.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+# Get the arguments
+args = parser.parse_args()
+
+# You need one (and only one) QApplication instance per application.
+app = QApplication([])
+
+# Create a Qt main window, which will be our window.
+window_main = MainWindow(args.verbose)
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply bin/run_m2gui.py:40]: The log message now is more like the debugging and learning the use of `pyside2` now (to let me know the code is working correctly). Most of log messages will be removed in the future after the code is in a good shape (I plan to take 2 - 3 months to finish this Python GUI). And I will keep your suggestions in mind at that time. Thanks!

File: python/lsst/ts/m2gui/utils.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+from PySide2.QtWidgets import QPushButton
+
+
+class LocalMode(IntEnum):
+    Standby = 1
+    Diagnostic = auto()
+    Enable = auto()
+
+
+def set_button(name, callback, *args, is_checkable=False):
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply python/lsst/ts/m2gui/utils.py:35]: I will add a class when I begin to add more functions. Right now, only this function is reused by multiple classes. Therefore, I put it into the `utils.py` as a placeholder at this moment. Thanks!

File: python/lsst/ts/m2gui/utils.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["set_button", "LocalMode"]
+
+from enum import IntEnum, auto
+from functools import partial
+from PySide2.QtWidgets import QPushButton
+
+
+class LocalMode(IntEnum):
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply python/lsst/ts/m2gui/utils.py]: Done in the commit: 1de0689. Thanks!

File: bin/run_m2gui.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+
+import argparse
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply bin/run_m2gui.py]: Done in the commit: 1de0689. Thanks!

**Pull request merged**